platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/platformstatus/statuslogic/ActiveStatusLogic.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.system.platformstatus.statuslogic;
+
+import com.swirlds.common.system.platformstatus.IllegalPlatformStatusException;
+import com.swirlds.common.system.platformstatus.PlatformStatus;
+import com.swirlds.common.system.platformstatus.PlatformStatusConfig;
+import com.swirlds.common.system.platformstatus.statusactions.CatastrophicFailureAction;
+import com.swirlds.common.system.platformstatus.statusactions.DoneReplayingEventsAction;
+import com.swirlds.common.system.platformstatus.statusactions.FallenBehindAction;
+import com.swirlds.common.system.platformstatus.statusactions.FreezePeriodEnteredAction;
+import com.swirlds.common.system.platformstatus.statusactions.ReconnectCompleteAction;
+import com.swirlds.common.system.platformstatus.statusactions.SelfEventReachedConsensusAction;
+import com.swirlds.common.system.platformstatus.statusactions.StartedReplayingEventsAction;
+import com.swirlds.common.system.platformstatus.statusactions.StateWrittenToDiskAction;
+import com.swirlds.common.system.platformstatus.statusactions.TimeElapsedAction;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Objects;
+
+/**
+ * Class containing the state machine logic for the {@link PlatformStatus#ACTIVE} status.
+ */
+public class ActiveStatusLogic implements PlatformStatusLogic {
+    /**
+     * The platform status config
+     */
+    private final PlatformStatusConfig config;
+
+    /**
+     * The last time a self event was observed reaching consensus
+     */
+    private Instant lastTimeOwnEventReachedConsensus;
+
+    /**
+     * Constructor
+     *
+     * @param startTime the time when the platform transitioned to the {@link PlatformStatus#ACTIVE} status
+     * @param config    the platform status config
+     */
+    public ActiveStatusLogic(@NonNull final Instant startTime, @NonNull final PlatformStatusConfig config) {
+        // a self event had to reach consensus to arrive at the ACTIVE status
+        this.lastTimeOwnEventReachedConsensus = Objects.requireNonNull(startTime);
```

Comment:
Since a transition to `ACTIVE` is always precipitated by a self event reaching consensus, the `startTime` is the initial value for `lastSelfEventConsensusTime`

I attempted to clarify this with the comment `a self event had to reach consensus to arrive at the ACTIVE status`, but I can rename the parameter if you prefer

Reply:
Naming has been adjusted to hopefully remove confusion